File: Valid anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashmapwannabe1={}
        hashmapwannabe2={}
        if(len(t)!=len(s)):
            return False
        for ele,ele2 in zip(t,s):
            hashmapwannabe1[ele]= hashmapwannabe1.get(ele, 0)+1
            hashmapwannabe2[ele2]= hashmapwannabe2.get(ele2, 0)+1
        if(hashmapwannabe1 == hashmapwannabe2):
                return True
        else:
            return False
                
# from collections import defaultdict

# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         if len(s) != len(t):
#             return False

#         count = defaultdict(int)
#         # Increment for s, decrement for t
#         for char_s, char_t in zip(s, t):
#             count[char_s] += 1
#             count[char_t] -= 1

#         # All zero counts means perfect anagram
#         return all(v == 0 for v in count.values())

File: Valid anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if(t.length != s.length)
    {
        return false;
    }
    const count = new Map();
    for(let i=0; i<t.length; i++)
        {
            const c1 = s[i]
            const c2 = t[i]
            count.set(c1, (count.get(c1) || 0) + 1);
            count.set(c2, (count.get(c2) || 0) - 1);
        }
    for (const v of count.values()) {
        if (v !== 0) return false;
    }
    // now that every count was zero:
    return true;
    
    
    };